### PR TITLE
Fix intro_inventory documented params. Update for the new 'become' system

### PR DESCRIPTION
**Pull request by jody-frankowski**
Replaces and updates #9588

### CODE REVIEW

**Comment by amenonsen** [comment]
Looks good, nice touch to include a link to the privilege escalation documentation too.

File: docsite/rst/intro_inventory.rst [hunk 1]
```diff
@@ -203,16 +204,38 @@ mentioned::
       The default ssh user name to use.
     ansible_ssh_pass
       The ssh password to use (this is insecure, we strongly recommend using --ask-pass or SSH keys)
+    ansible_connection
+      Connection type of the host. Candidates are local, ssh or paramiko.  The default is paramiko before Ansible 1.2, and 'smart' afterwards which detects whether usage of 'ssh' would be feasible based on whether ControlPersist is supported.
+    ansible_ssh_private_key_file
```
Comment (bcoca) [inline docsite/rst/intro_inventory.rst:212]: 1.2 is really old at this point, i would not even mention the change here

File: docsite/rst/intro_inventory.rst [hunk 2]
```diff
@@ -203,16 +204,38 @@ mentioned::
...
+      Private key file used by ssh.  Useful if using multiple keys and you don't want to use SSH agent.
+
+Privilege escalation (see :doc:`Ansible Privilege Escalation<become>` for further details). Since Ansible 1.9 you should use *ansible_become*::
+
```
Comment (bcoca) [inline docsite/rst/intro_inventory.rst:216]: we try to keep the intro light, so maybe just this link with a mention that other parameters exist might be a better change

Comment (jody-frankowski) [inline docsite/rst/intro_inventory.rst:216]: If we remove ansible_su/do from there I don't see anywhere else where they are documented.

Comment (bcoca) [inline docsite/rst/intro_inventory.rst:216]: we can adde them to the become page under 'legacy'

**Comment by amenonsen** [comment]
OK, this makes sense to me too.

**Pull request merged**